enterprise/internal/batches/store/changesets_test.go

```diff
@@ -2428,3 +2432,84 @@ func TestEnqueueChangesetsToClose(t *testing.T) {
 		ct.ReloadAndAssertChangeset(t, ctx, s, changeset, want)
 	}
 }
+
+func TestCleanDetachedChangesets(t *testing.T) {
+	logger := logtest.Scoped(t)
+	ctx := context.Background()
+	db := database.NewDB(logger, dbtest.NewDB(logger, t))
+
+	s := New(db, &observation.TestContext, nil)
+	rs := database.ReposWith(logger, s)
+	es := database.ExternalServicesWith(logger, s)
+
+	repo := ct.TestRepo(t, es, extsvc.KindGitHub)
+	err := rs.Create(ctx, repo)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name        string
+		cs          *btypes.Changeset
+		wantDeleted bool
+	}{
+		{
+			name: "old detached changeset deleted",
+			cs: &btypes.Changeset{
+				RepoID:              repo.ID,
+				ExternalID:          fmt.Sprintf("foobar-%d", 42),
+				ExternalServiceType: extsvc.TypeGitHub,
+				ExternalBranch:      "refs/heads/batch-changes/test",
+				// Set beyond the retention period
+				DetachedAt: time.Now().Add(time.Duration(-48) * time.Hour),
```

Comment:
This should be simplifiable to:

```suggestion
				DetachedAt: time.Now().Add(-48 * time.Hour),
```